pytorch_lightning/utilities/fetching.py

```diff
@@ -35,7 +38,21 @@
 class AbstractDataFetcher(ABC):
 
     """
-    This class is used to control batch fetching flow.
+    This based class should be used to implement a fault tolerant `DataFetcher`.
+
+    It is required to override the ``fetching_function`` with fetching logic.
```

Comment:
if this is required, we should remove the `pass` as the default implementation for the `fetching_function` in the `AbstractDataFetcher`

Reply:
Done !